quiche/src/tls.rs

```diff
@@ -1093,6 +1095,24 @@ fn map_result_ptr<'a, T>(bssl_result: *const T) -> Result<&'a T> {
     }
 }
 
+fn set_local_error_if_needed(
+    ssl: &mut Handshake, ex_data: &mut ExData, bssl_result: c_int,
+) {
+    // SSL_ERROR_SSL
+    if ssl.get_error(bssl_result) == 1 {
+        // SSL_ERROR_SSL can't be recovered so ensure we set a
+        // local_error so the connection is closed.
+        // See https://www.openssl.org/docs/man1.1.1/man3/SSL_get_error.html
+        if ex_data.local_error.is_none() {
+            *ex_data.local_error = Some(ConnectionError {
+                is_app: false,
+                error_code: TLS_ALERT_ERROR,
```

Comment:
This  is not actually a TLS alert though, so it should be `0x01` (`INTERNAL_ERROR`) instead IMO.